ci/docker/build.Dockerfile

```diff
@@ -0,0 +1,97 @@
+ARG BASE_IMAGE=ubuntu:20.04
+
+FROM $BASE_IMAGE
+
+# set jfrog autoclean policy
+LABEL com.jfrog.artifactory.retention.maxDays="21"
+
+ENV DEBIAN_FRONTEND=noninteractive \
+    PATH="$PATH:/opt/spack/bin:/opt/libtree" \
+    SPACK_COLOR=always
+SHELL ["/bin/bash", "-c"]
+
+RUN apt-get -yqq update && \
+    apt-get -yqq install --no-install-recommends \
+    software-properties-common \
+    build-essential gfortran \
+    autoconf automake \
+    clang \
+    gawk \
+    python3 python3.8-distutils \
+    git tar wget curl ca-certificates gpg-agent jq tzdata \
+    patchelf unzip file gnupg2 && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install cmake
+RUN wget -qO- "https://cmake.org/files/v3.22/cmake-3.22.1-linux-x86_64.tar.gz" | tar --strip-components=1 -xz -C /usr/local
+
+# Install libtree for packaging
+RUN mkdir -p /opt/libtree && \
+    curl -Lfso /opt/libtree/libtree https://github.com/haampie/libtree/releases/download/v2.0.0/libtree_x86_64 && \
+    chmod +x /opt/libtree/libtree
+
+# Install MKL and remove static libs (to keep image smaller)
+ARG USE_MKL=ON
+ARG MKL_VERSION=2020.4-912
+ARG MKL_SPEC=2020.4.304
+RUN if [ "$USE_MKL" = "ON" ]; then \
+      wget -qO - https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2019.PUB 2>/dev/null | apt-key add - && \
+      apt-add-repository 'deb https://apt.repos.intel.com/mkl all main' && \
+      apt-get install -y -qq --no-install-recommends intel-mkl-64bit-${MKL_VERSION} && \
+      rm -rf /var/lib/apt/lists/* && \
+      find "/opt/intel/" -name "*.a" -delete && \
+      echo -e "/opt/intel/lib/intel64\n/opt/intel/compilers_and_libraries/linux/mkl/lib/intel64" >> /etc/ld.so.conf.d/intel.conf && \
+      ldconfig ; \
+    fi
```

Comment:
What's the reason for not getting `intel-mkl` through spack?

Reply:
Image size (don't include 32 bit binaries). But I wonder if special-casing mkl is worth it.

Reply:
The size of the package. spack installs the full mkl (ia32 and intel64) while this method only installs intel64 libs.
This has two big advantages:
- less load on the network and filesystem when building the build image
- smaller build images (also achiavable with @haampie method)